aiven/resource_flink_table.go

```diff
@@ -0,0 +1,219 @@
+
+	project, serviceName, tableId := splitResourceID3(d.Id())
+
+	r, err := client.FlinkTables.Get(project, serviceName, aiven.GetFlinkTableRequest{TableId: tableId})
+	if err != nil {
+		return diag.FromErr(resourceReadHandleNotFound(err, d))
+	}
+
+	if err := d.Set("project", project); err != nil {
+		return diag.Errorf("error setting Flink Tables `project` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("service_name", serviceName); err != nil {
+		return diag.Errorf("error setting Flink Tables `service_name` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("integration_id", r.IntegrationId); err != nil {
+		return diag.Errorf("error setting Flink Tables `integration_id` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("table_id", r.TableId); err != nil {
+		return diag.Errorf("error setting Flink Tables `table_id` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("table_name", r.TableName); err != nil {
+		return diag.Errorf("error setting Flink Tables `table_name` for resource %s: %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceFlinkTableState(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	di := resourceFlinkTableRead(ctx, d, m)
+	if di.HasError() {
+		return nil, fmt.Errorf("cannot get flink table: %v", di)
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceFlinkTableCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	project := d.Get("project").(string)
+	serviceName := d.Get("service_name").(string)
+	integrationId := d.Get("integration_id").(string)
+	jdbcTable := d.Get("jdbc_table").(string)
+	kafkaTopic := d.Get("kafka_topic").(string)
+	likeOptions := d.Get("like_options").(string)
+	tableName := d.Get("table_name").(string)
+	partitionedBy := d.Get("partitioned_by").(string)
+	schemaSQL := d.Get("schema_sql").(string)
+
+	createRequest := aiven.CreateFlinkTableRequest{
+		IntegrationId: integrationId,
+		JDBCTable:     jdbcTable,
+		KafkaTopic:    kafkaTopic,
+		LikeOptions:   likeOptions,
+		Name:          tableName,
+		PartitionedBy: partitionedBy,
+		SchemaSQL:     schemaSQL,
+	}
+
+	conf := &resource.StateChangeConf{
+		Pending: []string{"CREATING"},
+		Target:  []string{"CREATED"},
+		Refresh: func() (interface{}, string, error) {
+			r, err := client.FlinkTables.Create(project, serviceName, createRequest)
+			if err != nil {
+				if aiven.IsNotFound(err) {
+					// if the integration could not be found we want to retry
+					return nil, "CREATING", nil
+				}
+				return nil, "", err
+			}
+			return r, "CREATED", nil
+		},
+		Delay:      1 * time.Second,
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 1 * time.Minute,
+	}
+
+	r, err := conf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tableId := r.(*aiven.CreateFlinkTableResponse).TableId
+	if err := d.Set("table_id", tableId); err != nil {
```

Comment:
Let's move that to the `resourceFlinkTableRead`

Reply:
:heavy_check_mark: